components/gantry/oneaxis/oneaxis.go

```diff
@@ -412,9 +479,12 @@ func (g *oneAxis) Stop(ctx context.Context, extra map[string]interface{}) error
 
 // Close calls stop.
 func (g *oneAxis) Close(ctx context.Context) error {
-	g.mu.Lock()
-	defer g.mu.Unlock()
-	return g.Stop(ctx, nil)
+	g.cancelFunc()
```

Comment:
Same trouble as before: I think `g.cancelFunc()` and `g.activeBackgroundWorkers.Wait()` should  go together, and _both_ should go after calling `g.Stop`. Otherwise, we can get into a situation where we stop the limit-switch-watcher before we stop the motor, and then the motor drives off the end of the gantry.